lambdas/indexer/app.py

```diff
@@ -74,27 +74,28 @@ def hello():
     return {'Hello': 'World!'}
 
 
+@app.route('/delete', methods=['POST'])
 @app.route('/', methods=['POST'])
 def post_notification():
     """
-    Receive a notification event and either queue it for asynchronous indexing or process it synchronously.
+    Receive a notification event and queue it for indexing or deletion.
     """
     hmac.verify(current_request=app.current_request)
     notification = app.current_request.json_body
     log.info("Received notification %r", notification)
     validate_request_syntax(notification)
-    params = app.current_request.query_params
+    if app.current_request.context['path'].endswith('/'):
```

Comment:
Not sure why this uses `endswith()`. Is it because the handler for `/health` uses it? The situation there is a little more intricate and warrants use of `endswith()`. Here however, I feel that we can just use `==`.

One thing to look out for is how Chalice treats a trailing slash. For example, both, 

https://service.explore.data.humancellatlas.org/repository/summary
https://service.explore.data.humancellatlas.org/repository/summary/

hit the summary endpoint even though it is bound to "/repository/summary" only. It's possible that Chalice strips the trailing slash so `app.current_request.context['path']` would be "/repository/summary" in both cases. It's also possible that it doesn't, and that would confuse your logic here. You'd need to experiment to find out.